core/app/models/spree/null_promotion_handler.rb

```diff
@@ -13,6 +13,10 @@ def activate
       @order
     end
 
+    def can_apply?
+      !@order.completed?
```

Comment:
Not sure we want any logic in here 🤔 

Maybe just return `true` like all other methods in here?

Reply:
The logic is really quite dumb, and since this is relevant for specs I would prefer it to do what it's supposed to be doing. If we just return `true`, we could add coupon codes to shipped orders, which I don't think makes sense.

Reply:
Sure, but maybe I don't understand the NullPromotionHandler then. Shouldn't it just justify the API. If it makes sense or not should be part of a concrete implementation, no?